pkg/download/internal/templatetools/template.go

```diff
@@ -18,13 +18,14 @@ package templatetools
 
 import (
 	"encoding/json"
+
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config/parameter/value"
 )
 
 type JSONObject map[string]any
 
 // NewJSONObject is a function that creates a JSONObject from a raw JSON byte slice.
-func NewJSONObject(raw []byte) (JSONObject, error) {
+func NewJSONObject(raw json.RawMessage) (JSONObject, error) {
```

Comment:
I am not sure why you changed this here to `json.RawMessage`.
Without this change, I would've approved the PR immediately, as it's straightforward enough. 

Since `json.RawMessage` is also not a type alias but a real type, I am really not sure if we should introduce it everywhere 🤔

Reply:
I agree - is there a need for this change? I believe the reason for `json.RawMessage` is be able to implement `Marshaler` and `Unmarshaler`.

Reply:
The usage of `json.RawMessage` emphasizes that the input should be JSON.

Could you explain the problems that defining an argument as `json.RawMessage` causes?